### PR TITLE
Added examples of using the plugin to compile into a Web Component html

### PR DESCRIPTION
**Pull request by homeslicesolutions**
I figured a way to make it to compile to a web component.  I think this is very useful for those who are working with web components!  Let me know what you think!

### CODE REVIEW

File: examples/web-component/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  module: {
+    loaders: [
+      { test: /\.css$/, loader: ExtractTextPlugin.extract('style-loader', 'css-loader') }
+    ]
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
+      inject: false,
+      cache: false,
```
Comment (jantimon) [inline examples/web-component/webpack.config.js]: why do you have to disable the cache?

Comment (homeslicesolutions) [inline examples/web-component/webpack.config.js]: There actually no need for that.  I just copied and pasted from one of the examples.  Sorry for the sloppiness.  No need to disable cache.

File: examples/web-component/template.ejs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<template>
+  <% htmlWebpackPlugin.files.css.forEach(function(cssFile) { %>
+  <style>@import "<%=cssFile%>";</style>
```
Comment (jantimon) [inline examples/web-component/template.ejs]: What about surrounding the loop with the style tag?

**Comment by jantimon** [comment]
Looks cool - the webpack 2 tests are missing right now.
Would this also work for multiple web components?

**Comment by homeslicesolutions** [comment]
@jantimon Yes there are ways to do that.  I can include that example as well.  I imagine the desire would have one importable HTML web component per entry point?

File: examples/web-component/template.ejs (hunk 2 above)
Comment (homeslicesolutions) [inline examples/web-component/template.ejs]: Touche.  Good look!

**Comment by homeslicesolutions** [comment]
@jantimon I have added new changes and an example of "multiple" web components per entry point

**Comment by homeslicesolutions** [comment]
@jantimon Do you know why these tests are failing one after the other?  Are these errors a false positive?

**Comment by jantimon** [comment]
You have to remove the webpack 1 module install webpack 2 module and build your new examples

**Comment by homeslicesolutions** [comment]
I see the issue, so the "semistandard" prepublish fails because it can't find "HTMLElement" a webcomponent feature.  How can I get it to ignore that check?  I tried environment targeting to browser but no fix.  This is getting sort of ridiculous just adding a few examples...

**Comment by jantimon** [comment]
Sorry :)
We use the examples for integration tests to ensure that every example will work after we changed the plugin.. Don't know what you mean by HTMLElement

**Comment by homeslicesolutions** [comment]
So in my examples, I am showing that you can use this plugin to bundle valid HTML Web Components.  Web components figure 4 new technologies that exists native-ly in a few popular browsers including Chrome and Firefox.  One main feature of Web Components is "Custom Element" which exposes an API class called "HTMLElement"  which you can extend and create your own custom element.  semistandard is using eslint or standard or something to make a check in the JS files and doesn't recognize "HTMLElement" as a valid class possibly there's a setting or the library is outdated, hence it fails.  How familiar are you with the semistandard library and are there ways to stub in the HTMLElement class?

**Comment by jantimon** [comment]
Just use `window.HTMLElement`.

**Comment by homeslicesolutions** [comment]
Wow ok! =)  it worked.  Be back with an update!

**Comment by homeslicesolutions** [comment]
I almost want to give up hahah =p.  What's going on?  So there's a Appcache failure in the test build.  And locally, i noticed that appcache isn't compatible with webpack 2.  Whats going on here?  Seems like a mess.

**Comment by homeslicesolutions** [comment]
image

**Comment by jantimon** [comment]
Welcome to my world ;)
Just ignore those errors - they work fine 👍

**Comment by homeslicesolutions** [comment]
alright so you bless this pull request? ;)

**Pull request closed without merging**